### PR TITLE
DriverSAM/NetworkInterface.c warning cleanup - purely refactoring

### PR DESCRIPTION
**Pull request by Peter-Herrmann**
DriverSAM/NetworkInterface.c warning cleanup refactor

Description
-----------

- Removed 6 unused variables to address [-Wunused-variable] warnings. One of these, `uxLowestSemCount` was also shadowing a file scope variable, which is misleading to a reader.
- Created function prototypes for 2 functions to address [-Wmissing-declarations]

Test Steps
-----------
The unused variables did not appear anywhere else in the library and are optimized out when compiling with -O1. With -O1 there is no difference with or without the unused variables.

The function prototypes do not appear in other files, so they are not violating MISRA C 2012 rule 8.5, and adding them satisfies rule 8.4. 

No new unit tests or modifications needed, this is purely a refactor. I have tested functionality within my application [here](https://github.com/Peter-Herrmann/SAME70-FreeRTOS-Plus-TCP).

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
No issue related

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/portable/NetworkInterface/DriverSAM/NetworkInterface.c [hunk 1]
```diff
@@ -1126,14 +1125,12 @@ void vNetworkInterfaceAllocateRAMToBuffers( NetworkBufferDescriptor_t pxNetworkB
 static void prvEMACHandlerTask( void * pvParameters )
 {
     UBaseType_t uxCount;
-    UBaseType_t uxLowestSemCount = GMAC_TX_BUFFERS + 1;
```
Comment (Peter-Herrmann) [inline source/portable/NetworkInterface/DriverSAM/NetworkInterface.c:1129]: This variable is unused, but it shadows a variable declared on line 1065. If this was intended to modify the global variable, it is not doing so. If it was not intended to modify the global variable, I'd definitely remove it.

File: source/portable/NetworkInterface/DriverSAM/NetworkInterface.c [hunk 2]
```diff
@@ -204,14 +204,17 @@ static void hand_tx_errors( void );
 static uint16_t prvGenerateCRC16( const uint8_t * pucAddress );
 static void prvAddMulticastMACAddress( const uint8_t * ucMacAddress );
 
+/* Checks IP queue, buffers, and semaphore and logs diagnostic info if configured */
+void vCheckBuffersAndQueue( void );
```
Comment (tony-josi-aws) [inline source/portable/NetworkInterface/DriverSAM/NetworkInterface.c:208]: I believe its better to keep `vCheckBuffersAndQueue` as a static function as its only used here.

File: source/portable/NetworkInterface/DriverSAM/NetworkInterface.c (hunk 1 above)
Comment (tony-josi-aws) [reply source/portable/NetworkInterface/DriverSAM/NetworkInterface.c:1129]: I believe it'll be better to update this line to modify the global variable - `uxLowestSemCount` for the diagnostic info checks in the `vCheckBuffersAndQueue` to work correctly.

**Comment by tony-josi-aws** [comment]
@Peter-Herrmann , thanks for taking time to contribute to FreeRTOS+TCP.

I have added some minor comments.

**Comment by Peter-Herrmann** [comment]
@tony-josi-aws thank you for the response. I will validate the changes you suggested tomorrow morning and update the PR.

**Review by htibosch — APPROVED** [review]
I agree with all cleanups, thank you for this PR.

File: source/portable/NetworkInterface/DriverSAM/NetworkInterface.c (hunk 1 above)
Comment (Peter-Herrmann) [reply source/portable/NetworkInterface/DriverSAM/NetworkInterface.c:1129]: Sounds good. Now the line updates the global.

File: source/portable/NetworkInterface/DriverSAM/NetworkInterface.c (hunk 2 above)
Comment (Peter-Herrmann) [reply source/portable/NetworkInterface/DriverSAM/NetworkInterface.c]: Agreed. Changed to a static funciton in [a715759](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1016/commits/a715759cfd6c6908f9ac183a04c6e954c9a72c99)

**Comment by Peter-Herrmann** [comment]
Thanks @htibosch! I've incorporated your feedback in [a715759](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1016/commits/a715759cfd6c6908f9ac183a04c6e954c9a72c99).

**Review by tony-josi-aws — DISMISSED** [review]

**Comment by tony-josi-aws** [comment]
Hi @Peter-Herrmann, thanks for updating the PR. The formatting CI checks are failing because #1014 needs to be merged first.

**Comment by Peter-Herrmann** [comment]
Looks like #1014 was merged in [eed294c](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/commit/eed294c0ec45433c85597623375a5dbf69058980). This PR merged in new changes in [c7d4106](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1016/commits/c7d4106952c4c09a687f93df08c4e67b0949259c).

**Review by tony-josi-aws — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**